clients/java/signalr/src/main/java/JsonHubProtocol.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class JsonHubProtocol implements HubProtocol {
+    private final JsonParser jsonParser = new JsonParser();
+    private final Gson gson = new Gson();
+    private static final String RECORD_SEPARATOR = "\u001e";
+
+    @Override
+    public String getName() {
+        return "json";
+    }
+
+    @Override
+    public int getVersion() {
+        return 1;
+    }
+
+    @Override
+    public TransferFormat getTransferFormat() {
+        return TransferFormat.Text;
+    }
+
+    @Override
+    public InvocationMessage[] parseMessages(String payload) {
+
+        String[] messages = payload.split(RECORD_SEPARATOR);
+        List<InvocationMessage> invocationMessages = new ArrayList<>();
+        // Empty handshake response "{}". We can ignore it
+        for (String splitMessage: messages) {
+            if (splitMessage.equals("{}")) {
+                continue;
+            }
+
+            JsonObject jsonMessage = jsonParser.parse(splitMessage).getAsJsonObject();
+            String messageType = jsonMessage.get("type").toString();
+            switch(messageType) {
```

Comment:
nit: space after `switch`

Reply:
looks like something a Java linter would detect and flag...